File: app/src/test/java/com/github/goldy1992/mp3player/dagger/components/TestServiceComponent.java
package com.github.goldy1992.mp3player.dagger.components;

import com.github.goldy1992.mp3player.dagger.modules.MockContentManagerModule;
import com.github.goldy1992.mp3player.dagger.modules.MockMediaSessionConnectorModule;
import com.github.goldy1992.mp3player.dagger.modules.service.ExoPlayerModule;
import com.github.goldy1992.mp3player.dagger.modules.service.HandlerThreadModule;
import com.github.goldy1992.mp3player.dagger.modules.service.MediaSessionCompatModule;
import com.github.goldy1992.mp3player.dagger.modules.service.ServiceModule;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {
        MockContentManagerModule.class,
        ExoPlayerModule.class,
        HandlerThreadModule.class,
        MockMediaSessionConnectorModule.class,
        MediaSessionCompatModule.class,
        ServiceModule.class })
public interface TestServiceComponent extends ServiceComponent {
    @Component.Factory
    interface Factory extends ServiceComponent.Factory {}
}

File: service/src/unitTest/java/com/github/goldy1992/mp3player/service/dagger/components/TestServiceComponent.kt
package com.github.goldy1992.mp3player.service.dagger.components

import com.github.goldy1992.mp3player.service.dagger.modules.MockSearchDatabaseModule
import com.github.goldy1992.mp3player.service.dagger.modules.service.*
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [
    ContentManagerModule::class,
    ContentRetrieversModule::class,
    ContentSearchersModule::class,
    ExoPlayerModule::class,
    MediaSessionConnectorModule::class,
    MediaSessionCompatModule::class,
    MockSearchDatabaseModule::class])
interface TestServiceComponent : ServiceComponent {
    @Component.Factory
    interface Factory : ServiceComponent.Factory
}